src/pages/SupplierRequisitionPage.js

```diff
@@ -96,56 +76,63 @@ export const SupplierRequisitionPage = ({ requisition, runWithLoadingIndicator,
     dataState,
     sortBy,
     isAscending,
-    columns,
     modalKey,
-    pageInfo,
     pageObject,
     hasSelection,
     showAll,
     keyExtractor,
     modalValue,
     searchTerm,
+    PageActions,
+    columns,
+    getPageInfoColumns,
   } = state;
 
   // Listen for changes to this pages requisition. Refreshing data on side effects i.e. finalizing.
-  useDatabaseChangeListener(dispatch, requisition, 'Requisition');
+  useRecordListener(() => dispatch(PageActions.refreshData()), requisition, 'Requisition');
 
   const { isFinalised, comment, theirRef, program } = pageObject;
 
   // On click handlers
-  const onCloseModal = () => dispatch(closeModal());
-  const onSelectItem = () => dispatch(openModal(MODAL_KEYS.SELECT_ITEM));
-  const onViewRegimenData = () => dispatch(openModal(MODAL_KEYS.VIEW_REGIMEN_DATA));
+  const onCloseModal = () => dispatch(PageActions.closeModal());
```

Comment:
For the functions which are passed down as props, should probably memoise?

Reply:
These are only passed to really basic components - i.e. buttons or the modal, which without a modal key is rendering `null` essentially, and doesn't re-render the page causing a new function to be created and propogate, so I don't really see the point (I'm trying to practise restraint, I really want to memo them :( ) - if it does need performance gains, we can add them but I think at that point we may aswell make the components class based. Also remember that there is stil the entire navigation stack mounted, and therefore all of the components memo'd data is still in the cach - at some point it's basically redundant to do so